examples/dmd/de_dg_advection_greedy.cpp

```diff
@@ -0,0 +1,1625 @@
+    args.AddOption(&greedy_param_space_f_factor_min, "-greedy-param-f-factor-min",
+                   "--greedy-param-f-factor-min",
+                   "The minimum f_factor value of the parameter point space.");
+    args.AddOption(&greedy_param_space_size, "-greedy-param-size",
+                   "--greedy-param-size",
+                   "The number of values to search in the parameter point space.");
+    args.AddOption(&greedy_relative_diff_tol, "-greedyreldifftol",
+                   "--greedyreldifftol", "The greedy algorithm relative diff tolerance.");
+    args.AddOption(&greedy_subset_size, "-greedysubsize", "--greedysubsize",
+                   "The greedy algorithm subset size.");
+    args.AddOption(&greedy_convergence_subset_size, "-greedyconvsize",
+                   "--greedyconvsize", "The greedy algorithm convergence subset size.");
+    // Differential Evolution Parameters
+    args.AddOption(&de, "-de", "--de",
+                   "-no-de", "--no-de",
+                   "Enable or disable the differential evolution phase.");
+    args.AddOption(&de_min_f_factor, "-de_min_ff", "--de_min_f-factor",
+                   "DE min f-factor");
+    args.AddOption(&de_max_f_factor, "-de_max_ff", "--de_max_f-factor",
+                   "DE max f-factor");
+    args.AddOption(&de_F, "-de_f", "--de_f",
+                   "DE F.");
+    args.AddOption(&de_CR, "-de_cr", "--de_cr",
+                   "DE CR.");
+    args.AddOption(&de_PS, "-de_ps", "--de_ps",
+                   "DE Population size.");
+    args.AddOption(&de_min_iter, "-de_min_iter", "--de_min_iter",
+                   "DE Minimum number of iterations.");
+    args.AddOption(&de_max_iter, "-de_max_iter", "--de_max_iter",
+                   "DE Maximum number of iterations.");
+    args.AddOption(&de_ct, "-de_ct", "--de_ct",
+                   "DE Cost threshold.");
+    args.Parse();
+    if (!args.Good())
+    {
+        if (mpi.Root())
+        {
+            args.PrintUsage(cout);
+        }
+        return 1;
+    }
+    if (mpi.Root())
+    {
+        args.PrintOptions(cout);
+    }
+
+    Device device(device_config);
+    if (mpi.Root()) {
+        device.Print();
+    }
+
+    io_dir = temp_io_dir;
+    mkdir(io_dir.c_str(), 0777);
+
+    // 3. Initialize the DMD database that will be built using a greedy algorithm.
+    if(de)
+    {
+        target_f_factor = f_factor;
+
+        simulation();
+
+        // Create relative error cost function in 4 dimensions (radius, alpha, cx, cy)
+        RelativeDifferenceCostFunction cost(1);
+
+        // Create Differential Evolution optimizer with population size of de_ps
+        // differential weight of de_F, and crossover probability of de_CR
+        CAROM::DifferentialEvolution de_opt(cost, de_PS, de_F, de_CR);
+
+        // Optimize for at least de_min_iter iterations, to a maximum of de_max_iter iterations with verbose output.
```

Comment:
Although it is not strictly enforced, we try to avoid long lines. In MFEM, "long" is 80 columns. In libROM, I don't think we have a maximum length. I would recommend 100 as a maximum maximum.